tests/components/unifiprotect/test_button.py

```diff
@@ -0,0 +1,73 @@
+"""Test the UniFi Protect button platform."""
+# pylint: disable=protected-access
+from __future__ import annotations
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from pyunifiprotect.data import Camera
+
+from homeassistant.components.unifiprotect.const import DEFAULT_ATTRIBUTION
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import MockEntityFixture, enable_entity
+
+
+@pytest.fixture(name="camera")
+async def camera_fixture(
+    hass: HomeAssistant, mock_entry: MockEntityFixture, mock_camera: Camera
+):
+    """Fixture for a single camera with only the button platform active, no extra setup."""
+
+    camera_obj = mock_camera.copy(deep=True)
+    camera_obj._api = mock_entry.api
+    camera_obj.channels[0]._api = mock_entry.api
+    camera_obj.channels[1]._api = mock_entry.api
+    camera_obj.channels[2]._api = mock_entry.api
+    camera_obj.name = "Test Camera"
+
+    mock_entry.api.bootstrap.cameras = {
+        camera_obj.id: camera_obj,
+    }
+
+    with patch("homeassistant.components.unifiprotect.PLATFORMS", [Platform.BUTTON]):
```

Comment:
Can we avoid patching the integration here ?

Reply:
Not easily. The patch makes this much easier to maintain so adding another entity in another platform will not affect the expected counts for this platform. I could iterate over all entities added and only pull out the ones for the selected platform, but patching the ones that gets initialized seems a lot cleaner to me.

Reply:
That would probably be a follow up PR though to do that since I am already doing the same pattern in camera and media_player.

Reply:
Ok. Let's save that for a follow up and do it in a separate PR.  Ideally we don't want to patch any of the integration code if we can avoid it